operations/fetch.go

```diff
@@ -205,6 +205,9 @@ type cloneOptions struct {
 
 func clone(opts cloneOptions) error {
 	// clone the repo first
+	if opts.token == "" {
```

Comment:
If `evergreen fetch` requires a token to be passed in now, doesn't that mean the `evergreen fetch` that Evergreen runs on their behalf also has to pass some kind of token? Did we test that spawning a host with task data still works as expected?

Reply:
I'm working on that for DEVPROD-8873. I thought I'd have the PR up already. Let me close this and re-open once ready.